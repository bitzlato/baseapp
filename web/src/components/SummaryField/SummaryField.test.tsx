import { shallow } from 'enzyme';
import * as React from 'react';
import { SummaryField, SummaryFieldProps } from './';

const defaultProps: SummaryFieldProps = {
  message: 'Message',
};

const setup = (props: Partial<SummaryFieldProps> = {}) =>
  shallow(
    <SummaryField {...{ ...defaultProps, ...props }}>
      <div>Content</div>
    </SummaryField>,
  );

describe('SummaryField', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have correct className', () => {
    const wrapper = setup();
    expect(wrapper.hasClass('cr-summary-field')).toBeTruthy();
  });

  it('should pass along supplied className', () => {
    const className = 'some-class';
    const wrapper = setup({ className });
    expect(wrapper.hasClass(className)).toBeTruthy();
  });

  it('should render correct message', () => {
    const wrapper = setup();
    const { children, className } = wrapper.find('span').first().props();
    expect(children).toContain('Message');
    expect(className).toContain('cr-summary-field-message');
  });
});
