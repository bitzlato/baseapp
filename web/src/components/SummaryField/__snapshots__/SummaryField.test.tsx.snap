// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryField should render 1`] = `
<ForwardRef
  className="cr-summary-field"
  row={true}
  spacing={true}
>
  <span
    className="cr-summary-field-message"
  >
    Message
  </span>
  <span
    className="cr-summary-field-content"
  >
    <div>
      Content
    </div>
  </span>
</ForwardRef>
`;
